python/ribasim/ribasim/model.py

```diff
@@ -80,42 +121,39 @@ class Model(BaseModel):
         Starting time of the simulation.
     endtime : Union[str, datetime.datetime]
         End time of the simulation.
-    allocation : Optional[Allocation]
-        Allocation settings.
     solver : Optional[Solver]
         Solver settings.
     logging : Optional[logging]
         Logging settings.
     """
 
-    node: Node
-    edge: Edge
-    basin: Basin
-    fractional_flow: Optional[FractionalFlow]
-    level_boundary: Optional[LevelBoundary]
-    flow_boundary: Optional[FlowBoundary]
-    linear_resistance: Optional[LinearResistance]
-    manning_resistance: Optional[ManningResistance]
-    tabulated_rating_curve: Optional[TabulatedRatingCurve]
-    pump: Optional[Pump]
-    outlet: Optional[Outlet]
-    terminal: Optional[Terminal]
-    discrete_control: Optional[DiscreteControl]
-    pid_control: Optional[PidControl]
-    user: Optional[User]
     starttime: datetime.datetime
     endtime: datetime.datetime
-    allocation: Optional[Allocation]
-    solver: Optional[Solver]
-    logging: Optional[Logging]
-
-    class Config:
-        validate_assignment = True
 
-    @classmethod
-    def fields(cls):
-        """Return the names of the fields contained in the Model."""
-        return cls.__fields__.keys()
+    update_timestep: float = 86400
+    relative_dir: str = "."
+    input_dir: str = "."
+    results_dir: str = "."
+
+    database: Database = Database()
+    allocation: Optional[Allocation] = Allocation()
+    results: Optional[Results] = Results()
+    solver: Optional[Solver] = Solver()
+    logging: Optional[Logging] = Logging()
```

Comment:
```suggestion
    allocation: Allocation | None = Allocation()
    results: Results | None = Results()
    solver: Solver | None = Solver()
    logging: Logging | None = Logging()
```

I am not sure how the `None` case would be triggered here